app/scripts/modules/core/src/managed/ManagedResourceStatusIndicator.tsx

```diff
@@ -177,30 +185,125 @@ const logClick = (label: string, resourceId: string, status: ManagedResourceStat
+          configuration and the current state of the resource.
+        </p>`;
+    body += viewConfigurationByStatus[status]?.pauseWarning ?? '';
+  } else {
+    body += `
+        <p>
+          Spinnaker will resume reconciling any differences between the declared configuration and the current state of
+          the resource.
+        </p>
+      `;
+  }
+  if (regions.length > 1) {
+    body += `
+        <p>
+          <div class="horizontal top sp-padding-m alert alert-warning">
+            <i class="fa fa-exclamation-triangle sp-margin-m-right sp-margin-xs-top"></i>
+            <span>${isPaused ? 'Resuming' : 'Pausing'} management of this resource will ${isPaused ? 'resume' : 'pause'}
+             management of this resource in the following managed regions: <b>${regions.join(', ')}</b>.
+            </span>
+          </div>
+        </p>`;
+  }
+  return body;
+};
+
+const LearnMoreLink = ({ resourceSummary }: { resourceSummary: IManagedResourceSummary }) => (
+  <a
+    target="_blank"
+    onClick={() => logClick('Status docs link', resourceSummary.id, resourceSummary.status)}
+    href={`https://www.spinnaker.io/reference/managed-delivery/resource-status/#${resourceSummary.status.toLowerCase()}`}
+  >
+    Learn more
+  </a>
+);
+
+const PopoverLinks = ({
+  resourceSummary,
+  application,
+  hidePopover,
+}: {
+  resourceSummary: IManagedResourceSummary;
+  application: Application;
+  hidePopover: () => void;
+}) => (
+  <div className="horizontal right">
+    {!resourceSummary.isPaused && (
+      <p className="sp-margin-m-top sp-margin-xs-bottom">
+        <button className="passive" onClick={() => toggleResourcePause(resourceSummary, application, hidePopover)}>
+          <i className="fa fa-pause" /> Pause management of this resource
+        </button>
+      </p>
+    )}
+    {resourceSummary.isPaused && !application.isManagementPaused && (
+      <p className="sp-margin-m-top sp-margin-xs-bottom">
+        <button className="passive" onClick={() => toggleResourcePause(resourceSummary, application, hidePopover)}>
+          <i className="fa fa-play" /> Resume management of this resource
+        </button>
+      </p>
+    )}
+    {application.isManagementPaused && (
+      <p>
+        <UISref to="home.applications.application.config" params={{ section: 'managed-resources' }}>
+          <a>Resume application management</a>
+        </UISref>
+      </p>
+    )}
+  </div>
 );
 
 export interface IManagedResourceStatusIndicatorProps {
   shape: 'square' | 'circle';
   resourceSummary: IManagedResourceSummary;
+  application: Application;
 }
 
 export const ManagedResourceStatusIndicator = ({
   shape,
-  resourceSummary: { id, status },
+  resourceSummary,
+  application,
 }: IManagedResourceStatusIndicatorProps) => {
+  const { status } = resourceSummary;
+  const PopoverContents = ({ hidePopover }: IHoverablePopoverContentsProps) => (
```

Comment:
Since this is used as a `ComponentType`, it think it makes more sense to define these contents as a functional component outside of the `ManagedResourceStatusIndicator`

Reply:
yep, agree, good call - will update

Reply:
I need to capture the the props and send them into the popover component, so I think I end up with something like:
```typescript
const getPopoverContents = (resourceSummary: IManagedResourceSummary, application: Application) => ({
  hidePopover,
}: IHoverablePopoverContentsProps) => (
  <>
    {viewConfigurationByStatus[resourceSummary.status].popoverContents(resourceSummary, application)}
    <PopoverLinks resourceSummary={resourceSummary} application={application} hidePopover={hidePopover} />
  </>
);
```
and then I'd have
```html
<HoverablePopover Component={getPopoverContents(resourceSummary, application)} placement="left">
```
It feels a little harder to read, but that could just be me, so I am happy to make the change or open to other suggestions on how to structure this.

Reply:
It looks like the HoverablePopover will apply the remaining props into the component that’s passed in. So if you add the props needed for the PopoverContents as props to HoverablePopover it will apply them. 

<HoverablePopover 
   Component={PopoverContents}
   resourceSummary={resourceSummary} />

Reply:
It does pass the props through, but the props for `HoverablePopover` are [defined here](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/presentation/HoverablePopover.tsx#L16-L46), and Typescript will throw an error if we add arbitrary properties to it, e.g.
```html
<HoverablePopover
        Component={PopoverContents}
        placement="left"
        resourceSummary={resourceSummary}
        application={application}
      >...
```
results in (roughly):
```
ERROR in ManagedResourceStatusIndicator.tsx(318,9): 
TS2322: ...Property 'resourceSummary' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<HoverablePopover>
```

Reply:
Alright then feel free to leave as is because no I agree that the code that works is the same level of readable as your original. Thanks for trying!

Reply:
from a readability perspective, I'm all for inlining. Separately, one thing that this will cause is a re-mount every time there's a render. With functional components, React decides how to re-mount versus just re-rendering based on reference identity — so by re-generating a new function here in every render we'll be un-mounting and re-mounting the entire DOM tree under PopoverContents (even though the other components you're rendering inside this one don't change their reference identity).

In my experience with APIs like the one `HoverablePopover` has right now, there's no way to get around this problem without changing the API itself (for the reasons you all have already discussed). Perhaps we should just have the `template` function pass in a `hidePopover` callback? Could also add a third prop (gross) like `render` that passes in args and is just a render prop instead of expecting a component.